core/src/main/scala/com/nvidia/spark/rapids/tool/Platform.scala

```diff
@@ -694,6 +699,7 @@ class EmrPlatform(gpuDevice: Option[GpuDevice],
   }
 
   override def isPlatformCSP: Boolean = true
+  override def maxGpusSupported: Int = 4
```

Comment:
same as previous comment about the origin of that value.

Reply:
Updated the logic to dynamically use the max value from the instance type map for the given platform else default to 1.